client/my-sites/upgrades/components/form/state-select.jsx

```diff
@@ -36,47 +36,56 @@ module.exports = React.createClass( {
 		} else {
 			this.refs.state.focus();
 		}
-	},
-
-	render: function() {
-		const classes = classNames( this.props.additionalClasses, 'state' ),
-			statesList = this.props.statesList.getByCountry( this.props.countryCode );
-		let options = [];
-
-		if ( isEmpty( statesList ) ) {
-			return (
-				<Input ref="state" { ...this.props } />
-			);
-		}
-
-		options.push( { key: '', label: this.translate( 'Select State' ), disabled: 'disabled' } );
+	}
 
-		options = options.concat( statesList.map( function( state ) {
-			if ( ! state.code ) {
-				return { key: '--', label: '', disabled: 'disabled' };
-			}
-			return { key: state.code, label: state.name };
-		} ) );
+	render() {
+		const classes = classNames( this.props.additionalClasses, 'state' );
 
 		return (
-			<div className={ classes }>
-				<div>
-					<FormLabel htmlFor={ this.props.name }>{ this.props.label }</FormLabel>
-					<FormSelect
-						ref="input"
-						name={ this.props.name }
-						value={ this.props.value }
-						disabled={ this.props.disabled }
-						onChange={ this.props.onChange }
-						onClick={ this.recordStateSelectClick }
-						isError={ this.props.isError } >
-						{ options.map( function( option ) {
-							return <option key={ option.key } value={ option.key } disabled={ option.disabled }>{ option.label }</option>;
-						} ) }
-					</FormSelect>
-				</div>
+			<div>
+				{ this.props.countryCode && <QueryCountryStates countryCode={ this.props.countryCode } /> }
+				{ isEmpty( this.props.countryStates )
+					? <Input ref="input" { ...this.props } />
+					: <div className={ classes }>
+						<FormLabel htmlFor={ this.props.name }>{ this.props.label }</FormLabel>
+						<FormSelect
+							ref="input"
+							name={ this.props.name }
+							value={ this.props.value }
+							disabled={ this.props.disabled }
+							onChange={ this.props.onChange }
+							onClick={ this.recordStateSelectClick.bind( this, this.props.eventFormName ) }
```

Comment:
We've enabled the [`react/jsx-no-bind` ESLint rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md) which this violates because `bind` will always create a new function, thereby preventing any possible benefits of pure render of a child component. Instead, you should bind components in the constructor:

```js
constructor() {
	super( ...arguments );

	this.recordStateSelectClick = this.recordStateSelectClick.bind( this );
}

recordStateSelectClick() {
	const { eventForm } = this.props;
	// ...
}
```

... or [stage 1 class instance fields](https://tc39.github.io/proposal-class-public-fields/):

```js
recordStateSelectClick = () => {
	const { eventForm } = this.props;
	// ...
}
```